Yoshi/QAKit/EnvironmentMenu/YoshiEnvironmentManager.swift

```diff
@@ -0,0 +1,19 @@
+//
+//  YoshiEnvironmentManager.swift
+//  Yoshi
+//
+//  Created by Kanglei Fang on 28/08/2017.
+//  Copyright © 2017 Prolific Interactive. All rights reserved.
+//
+
+import Foundation
+
+/// Environement manager contains current and all available environemnts info.
+public protocol YoshiEnvironmentManager: class {
```

Comment:
just curious why we're requiring the instance to be a `class`?

Reply:
Mostly because I want YoshiEnvironmentManager to be passed into the env menu and change the env by changing `YoshiEnvironmentManager`'s `currentEnvironment`,  this makes `YoshiEnvironmentManager` behave more like a class rather than a struct. Also, this helps prevent all the potential `mutating` keyword for whoever implements `YoshiEnvironmentManager`